docs/general/ddl/data-types.rst

```diff
@@ -2867,6 +2862,58 @@ constant array values.
 
 .. _data-types-geo:
 
+Nested arrays
+..............
```

Comment:
There are two hiccups here.

- Mismatch of section anchor `data-types-geo`.
- Length of section underline does not match section title.

I've submitted a patch at #13220.